src/curve/stable/StablePair.sol

```diff
@@ -113,73 +123,45 @@ contract StablePair is ReservoirPair {
         require(token0Fee <= type(uint112).max && token1Fee <= type(uint112).max, "SP: NON_OPTIMAL_FEE_TOO_LARGE");
     }
 
+    // TODO: Test re-entrancy.
+    // TODO: Should we use fallback?
     /// @dev Mints LP tokens - should be called via the router after transferring tokens.
     /// The router must ensure that sufficient LP tokens are minted by using the return value.
-    function mint(address to) public nonReentrant returns (uint256 liquidity) {
-        _syncManaged();
-
-        (uint112 lReserve0, uint112 lReserve1,) = getReserves();
-        (uint256 balance0, uint256 balance1) = _balance();
-
-        uint256 newLiq = _computeLiquidity(balance0, balance1);
-        uint256 amount0 = balance0 - lReserve0;
-        uint256 amount1 = balance1 - lReserve1;
-
-        (uint256 fee0, uint256 fee1) = _nonOptimalMintFee(amount0, amount1, lReserve0, lReserve1);
-        lReserve0 += uint112(fee0);
-        lReserve1 += uint112(fee1);
-
-        (uint256 _totalSupply, uint256 oldLiq) = _mintFee(lReserve0, lReserve1);
+    function mint(address) public returns (uint256) {
+        // DELEGATE TO StableMintBurn
+        address lTarget = MINT_BURN_LOGIC;
+        assembly {
+            calldatacopy(0, 0, calldatasize())
+            let success := delegatecall(gas(), lTarget, 0, calldatasize(), 0, 0)
+
+            if success {
+                returndatacopy(0, 0, returndatasize())
+                return(0, returndatasize())
+            }
 
-        if (_totalSupply == 0) {
-            require(amount0 > 0 && amount1 > 0, "SP: INVALID_AMOUNTS");
-            liquidity = newLiq - MINIMUM_LIQUIDITY;
-            _mint(address(0), MINIMUM_LIQUIDITY);
-        } else {
-            liquidity = ((newLiq - oldLiq) * _totalSupply) / oldLiq;
+            returndatacopy(0, 0, returndatasize())
+            revert(0, returndatasize())
         }
-        require(liquidity != 0, "SP: INSUFFICIENT_LIQ_MINTED");
-        _mint(to, liquidity);
-        _update(balance0, balance1, lReserve0, lReserve1);
-
-        // casting is safe as the max invariant would be 2 * uint112 * uint60 (in the case of tokens with 0 decimal
-        // places)
-        // which results in 112 + 60 + 1 = 173 bits
-        // which fits into uint192
-        // TODO: Why does lastInvariant get set to newLiq? Not symmetric with burn?
-        lastInvariant = uint192(newLiq);
-        lastInvariantAmp = _getCurrentAPrecise();
-
-        emit Mint(msg.sender, amount0, amount1);
-
-        _managerCallback();
     }
 
+    // TODO: Test re-entrancy.
+    // TODO: Should we use fallback?
```

Comment:
as in `fallback() external {}` ? Is the motivation gas savings? Even if we use it, we can't use it for both `mint` and `burn`, only one at most right?